store-api-server/src/payment/controller/payment.controller.ts

```diff
@@ -97,6 +98,8 @@ export class PaymentController {
       // temporary for backwards compatibility, until frontend has been updated
       paymentProvider = PaymentProvider.TEZPAY;
     }
+    const { ip } = request;
+    const clientIp = request.get('X-Forwarded-For') || ip;
```

Comment:
Slightly more correct to do this instead like it is done in https://github.com/umitduran/motion_kanvas/blob/feat/simplexapiintegration/store-api-server/src/middleware/logger.ts#L51. Because if the setup does not involve some kind of proxy ontop (like nginx), X-Forwarded-For is not a trusted request header (it can be set to whatever by the API caller then).

Not a huge deal in this case I think but I would still prefer to do this right. Only check the `X-Forwarded-For` header if the `BEHIND_PROXY` constant is true.

Reply:
In fact, would be nice to create a helper function for this in `utils.ts` and use that function in the middleware/logger.ts and here in payment.controller.ts

Reply:
actually firstly I tried to add BEHIND_PROXY enum But payment controller gives a build error. Then I have to remove this enum. I do not know how to imported. I guess it is enough to yarn start to be imported enums. :/ 

`Error: Cannot find module '../motion_kanvas/store-api-server/dist/constants' imported from /Users/hakan.kocatepe/PersonalWebStormProjects/kanvasfinal/motion_kanvas/store-api-server/dist/payment/controller/payment.controller.js`

Reply:
actually firstly I tried to add BEHIND_PROXY enum But payment controller gives a build error. Then I have to remove this enum. I do not know how to imported. I guess it is enough to yarn start to be imported enums. :/ 

`Error: Cannot find module '../motion_kanvas/store-api-server/dist/constants' imported from '.../motion_kanvas/store-api-server/dist/payment/controller/payment.controller.js`

Reply:
Few things:
- A compiler error just means there's some mistake in the way you tried to use BEHIND_PROXY? Doesn't mean that it is not possible. This error here looks to me like you made a mistake in the path to constants import
- BEHIND_PROXY is not an enum, it's a boolean value

Reply:
I caught error. because I imported ts file. Ah typescript :) I am moving to util.ts :)

Reply:
Typescript definitely is a bit weird with import paths yeah.. I really don't understand why they went with this "you have to import with .js extension" approach. I know of no other language that requires you to import based on the compiler's target language output.